### PR TITLE
Statsd support

### PR DESCRIPTION
**Pull request by reimannf**
I am not sure about the counters within the Scraper and TransferThreads. What do you think?

### CODE REVIEW

**Comment by reimannf** [comment]
https://github.com/sapcc/swift-http-import/pull/9

File: config.go [hunk 1]
```diff
@@ -55,6 +55,11 @@ type Configuration struct {
 	WorkerCounts struct {
 		Transfer uint
 	} `yaml:"workers"`
+	Statsd struct {
+		HostName 	string 	`yaml:"host_name"`
+		Port	 	string 	`yaml:"port"`
```
Comment (majewsky) [inline config.go]: `int` please

File: config.go [hunk 2]
```diff
@@ -55,6 +55,11 @@ type Configuration struct {
 	WorkerCounts struct {
 		Transfer uint
 	} `yaml:"workers"`
+	Statsd struct {
+		HostName 	string 	`yaml:"host_name"`
```
Comment (majewsky) [inline config.go]: `host_name` looks weird. That word is usually written as `hostname`.

File: README.md [hunk 3]
```diff
@@ -98,3 +98,28 @@ INFO: 4 dirs scanned; 45 files found, 0 transferred, 0 failed
...
+  port:     8125
+  prefix:   swift_http_import
+```
+
+The following metric are sent:
+
+| Kind    | Name                      |
+| ------- | ------------------------- |
+| Gauge   | last_run.success          |
+| Gauge   | last_run.duration_s       |
```
Comment (majewsky) [inline README.md]: Don't be afraid to use descriptive names: `duration_s -> duration_seconds`

File: util.go [hunk 4]
```diff
@@ -65,3 +66,19 @@ func Log(level LogLevel, msg string, args ...interface{}) {
...
+	}
+	return nil
+}
+
+func Inc(bucket string, value int64, rate float32) error {
+	if statsd_client != nil {
+		return statsd_client.Inc(bucket, value, rate)
+	}
+	return nil
+}
```
Comment (majewsky) [inline util.go]: Are the `statsd_client.Gauge/Inc` methods documented as thread-safe? If not, we should guard these functions with a mutex.

File: config.go (hunk 2 above)
Comment (reimannf) [inline config.go]: done

File: config.go (hunk 1 above)
Comment (reimannf) [inline config.go]: done

File: README.md (hunk 3 above)
Comment (reimannf) [inline README.md]: done

File: util.go (hunk 4 above)
Comment (reimannf) [inline util.go]: No Counters within threads anymore - was doubtful info anyhow

File: config.go [hunk 5]
```diff
@@ -89,6 +94,15 @@ func ReadConfiguration() (*Configuration, []error) {
 		cfg.WorkerCounts.Transfer = 1
 	}
 
+	// set default statsd port
+	if cfg.Statsd.HostName != "" && cfg.Statsd.Port == "" {
+		cfg.Statsd.Port = "8125"
```
Comment (majewsky) [inline config.go]: Please check if this compiles. `Port` is an int now.

**Comment by reimannf** [comment]
Closed with https://github.com/sapcc/swift-http-import/issues/6

**Pull request merged**